xarray/tests/test_groupby.py

```diff
@@ -3200,6 +3200,16 @@ def test_multiple_grouper_unsorted_order() -> None:
     assert_identical(actual2, expected2)
 
 
+def test_multiple_grouper_groups() -> None:
+    ds = xr.Dataset(
+        {"foo": (("x", "y"), np.random.rand(4, 3))},
+        coords={"x": [10, 20, 30, 40], "letters": ("x", list("abba"))},
+    )
+
+    groups = ds.groupby(x=BinGrouper(bins=[5, 15, 25]), letters=UniqueGrouper())
+    assert len(groups.groups) == 2
```

Comment:
this isn't right; there should be 4?

`([5,15); 'a'); ([5, 15); 'b'); ([15, 25), 'a'), ([15, 25), 'b')`.

Both of these groupers should have group_indices